source/speech/__init__.py

```diff
@@ -458,6 +474,34 @@ def _flattenNestedSequences(nestedSequences: Iterator[SpeechSequence]) -> Iterat
 	return [i for seq in nestedSequences for i in seq]
 
 
+def _getCancellableEvent(

```

Comment:
It feels a bit odd that this is in the speech module, checking focus and such. May be some of this logic can be moved to eventHandler, so speech only has to query eventHandler and doesn't have to import NVDAObject here?